runtime/vm/zcinterp.m4

```diff
@@ -153,70 +153,101 @@ END_CURRENT
 })
 
 ifdef({OMR_GC_CONCURRENT_SCAVENGER},{
-dnl Helper to handle guarded storage events and software read barriers. When
-dnl the event occurs, hardware/software populates the return address in 
-dnl J9VMThread.gsParameters.returnAddr.
-define({HANDLE_GS_EVENT_HELPER},{
-    SAVE_ALL_REGS($1)
-    ST_GPR J9SP,J9TR_VMThread_sp(J9VMTHREAD)
-    LR_GPR CARG1,J9VMTHREAD
-    L_GPR CRA,J9TR_VMThread_javaVM(J9VMTHREAD)
-    L_GPR CRA,J9TR_JavaVM_invokeJ9ReadBarrier(CRA)
-    CALL_INDIRECT(CRA)
-    L_GPR J9SP,J9TR_VMThread_sp(J9VMTHREAD)
-    ST_GPR J9SP,JIT_GPR_SAVE_SLOT(J9SP)
-    RESTORE_ALL_REGS_AND_SWITCH_TO_JAVA_STACK($1)
-})
+    define({HANDLE_HARDWARE_READ_BARRIER},{
+    BEGIN_HELPER($1)
+        SAVE_ALL_REGS($1)
 
-define({HANDLE_HARDWARE_READ_BARRIER},{
-BEGIN_FUNC($1)
 dnl Test the TX bit of GSECI (bit 16) to see if the CPU was in transactional-
```

Comment:
Should fix up the comment to say we test both TX and CTX bits of GSECI (bit 16 and 17).

Reply:
Fixed in f95c0c2